mod_modular_vanilla/hooks/skills/skill.nut

```diff
@@ -1,3 +1,38 @@
+::ModularVanilla.QueueBucket.VeryLate.push(function() {
+	::ModularVanilla.MH.hookTree("scripts/skills/skill", function(q) {
+		// This check exists because `skill` does not contain a `create` function.
+		if (q.ClassName != "skill")
+		{
+			// VanillaFix: https://steamcommunity.com/app/365360/discussions/1/604169856012736554/
+			// Vanilla does not update the `m.HitChanceBonus` with respect to the actual MeleeSkill modifier
+			// leading to wrong information in the Hitfactors tooltip.
+			// We set HitChanceBonus always to 0 as base value. Then set it to the correct value at the end
+			// of the onAnySkillUsed function.
+			q.create = @(__original) { function create()
+			{
+				__original();
+				this.m.HitChanceBonus = 0;
+			}}.create;
+
+			q.onAnySkillUsed = @(__original) { function onAnySkillUsed( _skill, _targetEntity, _properties )
+			{
+				if (_skill != this)
+				{
+					__original(_skill, _targetEntity, _properties);
+					return;
+				}
+
+				local mskill = _properties.MeleeSkill;
+				local bonus = this.m.HitChanceBonus;
+
+				__original(_skill, _targetEntity, _properties);
+
+				this.m.HitChanceBonus = bonus - mskill + _properties.MeleeSkill;
```

Comment:
Why are we preserving the previous `this.m.HitChanceBonus` value? I assumed from the mission statement of this PR that we are now the one source of truth for `HitChanceBonus` and don't care about any other calculation/logic coming before us